src/thriveopedia/ThriveopediaPage.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Godot;
+
+/// <summary>
+///   A page that can be opened in the Thriveopedia.
+/// </summary>
+public abstract class ThriveopediaPage : PanelContainer
+{
+    /// <summary>
+    /// Whether this page should display the default panel background.
+    /// </summary>
+    [Export]
+    public bool DisplayBackground = true;
+
+    private PanelContainer backgroundPanel = null!;
+
+    /// <summary>
+    ///   Details for the game currently in progress. Null if opened from the main menu.
+    /// </summary>
+    private GameProperties? currentGame;
+
+    /// <summary>
+    ///   The internal name of this page.
+    /// </summary>
+    public abstract string PageName { get; }
+
+    /// <summary>
+    ///   The translated name of this page.
+    /// </summary>
+    public abstract string TranslatedPageName { get; }
+
+    /// <summary>
+    ///   Details for the game currently in progress. Null if opened from the main menu. When set, runs any
+    ///   page-specific logic relating to the new game details.
+    /// </summary>
+    public GameProperties? CurrentGame
+    {
+        get => currentGame;
+        set
+        {
+            currentGame = value;
+
+            UpdateCurrentWorldDetails();
+        }
+    }
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        backgroundPanel = GetNode<PanelContainer>(".");
```

Comment:
Isn't background panel then just "this"? So on line 54 it should just be "this.AddStyle..."?